--- meson-0.62.2/mesonbuild/mesonmain.py.orig	2022-06-02 18:16:59.000000000 +0200
+++ meson-0.62.2/mesonbuild/mesonmain.py	2022-06-19 08:53:19.101307800 +0200
@@ -193,15 +193,18 @@
                   'regencheck': 'regen_checker'}
     module_name = script_map.get(script_name, script_name)
 
+    print("import_module")
     try:
         module = importlib.import_module('mesonbuild.scripts.' + module_name)
     except ModuleNotFoundError as e:
         mlog.exception(e)
         return 1
 
+    print("module.run", module)
     try:
         return module.run(script_args)
     except MesonException as e:
+        print("module.run: error")
         mlog.error(f'Error in {script_name} helper script:')
         mlog.exception(e)
         return 1
@@ -233,6 +236,7 @@
     # locale isn't Unicode, just force stdout to accept it. This tries
     # to emulate enough of PEP 540 to work elsewhere.
     ensure_stdout_accepts_unicode()
+    print("ensure_stdout_accepts_unicode")
 
     # https://github.com/mesonbuild/meson/issues/3653
     if sys.platform.lower() == 'msys':
@@ -246,6 +250,7 @@
 
     # Set the meson command that will be used to run scripts and so on
     mesonlib.set_meson_command(mainfile)
+    print("set_meson_command")
 
     args = original_args[:]
 
@@ -257,6 +262,7 @@
             # "meson --reconfigure"
             args = ['--reconfigure'] + args[2:]
         else:
+            print("run_script_command:")
             return run_script_command(args[1], args[2:])
 
     return CommandLineParser().run(args)
@@ -271,4 +277,6 @@
     return run(sys.argv[1:], launcher)
 
 if __name__ == '__main__':
-    sys.exit(main())
+    res = main()
+    print("DONEEEEE", res)
+    sys.exit(res)
