diff --git a/Python/pythonrun.c b/Python/pythonrun.c
index 0f1794a..d291fcd 100644
--- a/Python/pythonrun.c
+++ b/Python/pythonrun.c
@@ -453,8 +453,10 @@ _PyRun_SimpleFileObject(FILE *fp, PyObject *filename, int closeit,
             ret = -1;
             goto done;
         }
+        fprintf(stderr, "Before running the file, at Python/pythonrun.c:%d", __LINE__);
         v = pyrun_file(fp, filename, Py_file_input, d, d,
                        closeit, flags);
+        fprintf(stderr, "After running the file, at Python/pythonrun.c:%d", __LINE__);
     }
     flush_io();
     if (v == NULL) {
