diff --git a/Python/pythonrun.c b/Python/pythonrun.c
index 0f1794a..74b4091 100644
--- a/Python/pythonrun.c
+++ b/Python/pythonrun.c
@@ -453,8 +453,13 @@ _PyRun_SimpleFileObject(FILE *fp, PyObject *filename, int closeit,
             ret = -1;
             goto done;
         }
+        FILE* f_log;
+        f_log = fopen("C:\\python_mingw.log", "a");
+        fprintf(f_log, "Before running the file, at Python/pythonrun.c:%d\n", __LINE__);
         v = pyrun_file(fp, filename, Py_file_input, d, d,
                        closeit, flags);
+        fprintf(f_log, "After running the file, at Python/pythonrun.c:%d\n", __LINE__);
+        fclose(f_log);
     }
     flush_io();
     if (v == NULL) {
