diff --git a/Python/pythonrun.c b/Python/pythonrun.c
index 0f1794a..5915447 100644
--- a/Python/pythonrun.c
+++ b/Python/pythonrun.c
@@ -455,6 +455,7 @@ _PyRun_SimpleFileObject(FILE *fp, PyObject *filename, int closeit,
         }
         v = pyrun_file(fp, filename, Py_file_input, d, d,
                        closeit, flags);
+        fprintf(stderr, "After running the file, at Python/pythonrun.c:%d\n", __LINE__);
     }
     flush_io();
     if (v == NULL) {
