diff --git a/Modules/main.c b/Modules/main.c
index 9ee34a3..770cbbb 100644
--- a/Modules/main.c
+++ b/Modules/main.c
@@ -685,6 +685,7 @@ Py_RunMain(void)
 static int
 pymain_main(_PyArgv *args)
 {
+    int j, ret;
     PyStatus status = pymain_init(args);
     if (_PyStatus_IS_EXIT(status)) {
         pymain_free();
@@ -694,7 +695,28 @@ pymain_main(_PyArgv *args)
         pymain_exit_error(status);
     }
 
-    return Py_RunMain();
+    FILE *f_log;
+    f_log = fopen ("C:\\python_mingw.log", "a");
+    if (f_log == NULL)
+      {
+        perror ("Oh Well");
+        return 1;
+      }
+    fprintf (f_log, "Command line args are: \n");
+    for (j = 0; j < args->argc; j++)
+      {
+        if (args->use_bytes_argv == 1){
+            fprintf (f_log, "argv[%d]=%s\n", j, (args->bytes_argv)[j]);
+        }  else {
+            fprintf (f_log, "argv[%d]=%ls\n", j, (args->wchar_argv)[j]);
+        }
+      }
+    ret = Py_RunMain ();
+    fprintf (f_log, "Exiting with code: %d\n\n", ret);
+    fprintf (f_log, "\n\n\n\n");
+    fclose (f_log);
+
+    return ret;
 }
 
 
diff --git a/Python/pythonrun.c b/Python/pythonrun.c
index 0f1794a..0a35759 100644
--- a/Python/pythonrun.c
+++ b/Python/pythonrun.c
@@ -453,8 +453,13 @@ _PyRun_SimpleFileObject(FILE *fp, PyObject *filename, int closeit,
             ret = -1;
             goto done;
         }
+        FILE* f_log;
+        f_log = fopen("C:\\python_mingw.log", "w");
+        fprintf(f_log, "Before running the file, at Python/pythonrun.c:%d filename: %s\n", __LINE__, filename);
         v = pyrun_file(fp, filename, Py_file_input, d, d,
                        closeit, flags);
+        fprintf(f_log, "After running the file, at Python/pythonrun.c:%d\n", __LINE__);
+        fclose(f_log);
     }
     flush_io();
     if (v == NULL) {
