diff --git a/Python/frozenmain.c b/Python/frozenmain.c
index 6607762..29e6316 100644
--- a/Python/frozenmain.c
+++ b/Python/frozenmain.c
@@ -107,7 +107,7 @@ Py_FrozenMain(int argc, char **argv)
     }
     else
         sts = 0;
-
+    fprintf(stderr, "test before is_cygpty\n");
     if (inspect && (isatty((int)fileno(stdin)) || is_cygpty((int)fileno(stdin))))
         sts = PyRun_AnyFile(stdin, "<stdin>") != 0;
 
